code/datums/callback.dm

```diff
@@ -54,37 +56,48 @@
 	if (length(args) > 2)
 		arguments = args.Copy(3)
 
+/proc/ImmediateInvokeAsync(thingtocall, proctocall, ...)
+	set waitfor = FALSE
 
-/datum/callback/proc/Invoke(...)
-	if (!object)
+	if (!thingtocall)
 		return
 
-	var/list/calling_arguments = arguments
+	var/list/calling_arguments = length(args) > 2 ? args.Copy(3) : null
+
+	if (thingtocall == GLOBAL_PROC)
+		call(proctocall)(arglist(calling_arguments))
+	else
+		call(thingtocall, proctocall)(arglist(calling_arguments))
+
+#define CALLBACK_INVOKE \
+	if (!object)\
+	{\
+		return;\
+	}\
+	var/list/calling_arguments = arguments;\
+	if (length(args))\
+	{\
+		if (length(arguments))\
+		{\
+			calling_arguments = calling_arguments + args; /*not += so that it creates a new list so the arguments list stays clean*/\
+		}\
+		else\
+		{\
+			calling_arguments = args;\
+		}\
```

Comment:
What kind of messed up resolution are you running? because with 1680x1050 the Allman style macro already fits in half a screen, more than acceptable.

Also, don't call it "the normal style", K&R has never been the standard,, ever, where as Allman jointly ruled C/C++ with Whitesmith, which is just Allman but the braces are also indented, Allman is also the standard for C#

Braces having their own line, be they Whitesmith or Allman have been show to increase reading comprehension, however I can't actually find any studies for you on this because they were done when C and C++ were still kings.